docs/development.md

```diff
@@ -0,0 +1,342 @@
+
+# Run with backtrace for errors
+RUST_BACKTRACE=1 cargo run
+```
+
+## Project Structure
+
+```
+xtm-composer/
+├── src/
+│   ├── main.rs              # Application entry point
+│   ├── api/                 # External API integrations
+│   │   ├── opencti/         # OpenCTI API client
+│   │   └── openbas/         # OpenBAS API client (Coming Soon)
+│   ├── config/              # Configuration management
+│   │   ├── mod.rs
+│   │   └── settings.rs      # Settings structure definitions
+│   ├── engine/              # Core business logic
+│   │   ├── mod.rs
+│   │   └── manager.rs       # Manager implementation
+│   └── orchestrator/        # Container orchestration
+│       ├── mod.rs
+│       ├── kubernetes.rs    # Kubernetes implementation
+│       ├── docker.rs        # Docker implementation
+│       └── portainer.rs     # Portainer implementation
+├── config/                  # Configuration files
+│   ├── default.yaml         # Default configuration
+│   └── development.yaml     # Development overrides
+├── Cargo.toml              # Rust dependencies
+└── build.rs                # Build script
+```
+
+## Development Workflow
+
+### 1. Feature Development
+
+```bash
+# Create feature branch
+git checkout -b feature/your-feature-name
+
+# Make changes and test
+cargo build
+cargo test
+
+# Format code
+cargo fmt
+
+# Run linter
+cargo clippy -- -D warnings
+
+# Commit changes
+git add .
+git commit -m "feat: add your feature description"
+```
+
+### 2. Building
+
+```bash
+# Development build (with debug symbols)
+cargo build
+
+# Release build (optimized)
+cargo build --release
+
+# Check compilation without building
+cargo check
+```
+
+## Code Style Guidelines
+
+### Rust Best Practices
+
+1. **Follow Rust naming conventions**:
+   - Use `snake_case` for functions and variables
+   - Use `PascalCase` for types and traits
+   - Use `SCREAMING_SNAKE_CASE` for constants
+
+2. **Error Handling**:
+   - Use `Result<T, E>` for recoverable errors
+   - Use `panic!` only for unrecoverable errors
+   - Provide meaningful error messages
+
+3. **Documentation**:
+   - Add doc comments (`///`) for public APIs
+   - Include examples in documentation
+   - Document panics and errors
+
+### Example Code Style
+
+```rust
+/// Manages connector lifecycle in OpenCTI
+///
+/// # Examples
+///
+/// ```
+/// let manager = ConnectorManager::new(config)?;
+/// manager.start()?;
+/// ```
+pub struct ConnectorManager {
+    config: ManagerConfig,
+    orchestrator: Box<dyn Orchestrator>,
+}
+
+impl ConnectorManager {
+    /// Creates a new connector manager
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if configuration is invalid
+    pub fn new(config: ManagerConfig) -> Result<Self, ManagerError> {
+        // Implementation
+    }
+}
+```
+
+## Testing with Local Services
+
+### Running Local OpenCTI
+
+```bash
+# Clone OpenCTI
+git clone https://github.com/OpenCTI-Platform/opencti.git
+cd opencti/docker
```

Comment:
is it new ? I think the path is https://github.com/OpenCTI-Platform/opencti/tree/master/opencti-platform/opencti-dev
```suggestion
cd opencti/opencti-platform/opencti-dev
```